### ISSUE TITLE
Modernize design

### ISSUE THREAD

**Issue by ThatOneCalculator** [woheller69/Level#20]
While this app is very useful, the UI feels very dated. Bringing it up to MD3 design guidelines would help it feel cohesive with the rest of a modern Android phone.

**Comment by konfrag4** [comment]
Hi! I am not a professional UI designer, but I created this design based on MD3. The colors can be either static or dynamic based on the user's wallpaper (currently I use the default static color scheme for light mode). I haven't created one for the ruler and for when the phone is rotated horizontally yet. If you like my design or have any suggestions please let me know. (Obviously the design needs more polish, but there is no point in polishing a design that nobody likes)

![Level - Bubble Android App](https://github.com/woheller69/Level/assets/92216292/517b8bfd-95cc-452b-b5ce-633d58c7cd3b)

Thanks.

**Comment by ThatOneCalculator** [comment]
Even though the colors are material you, this design doesn't fit material design 3. I'd recommend looking at https://m3.material.io

**Comment by VasilisKos** [comment]
Definitely, a Material You design would make this app awesome. It's a pity to have such an outdated UI.

**Comment by woheller69** [comment]
Please try this. I reduced the number of colors. Should look a bit more modern now.

-Uninstall
-Remove .zip and install

[app-release.apk.zip](https://github.com/woheller69/Level/files/15049193/app-release.apk.zip)

![Screenshot_20240420-193251_Bubble](https://github.com/woheller69/Level/assets/68678880/ea190563-e400-4c54-8b69-e0866a01e938)

**Comment by ThatOneCalculator** [comment]
Having less colors doesn't really modernize a design. I'd recommend reading over https://m3.material.io/

**Comment by woheller69** [comment]
not going to waste time on that. Maybe someone wants to provide a PR

**Comment by VasilisKos** [comment]
A monochrome UI with a color accent taken from the wallpaper would be perfect.

**Comment by VasilisKos** [comment]
> not going to waste time on that. Maybe someone wants to provide a PR

I wouldn't use the word "waste". "Spend" would be much better... 🤨

**Comment by ig9000** [comment]
On a practical level, I would love to have better contrast for the rulers (and not by shining a glaring white screen in my face). A full black background mode would make a lot of sense with modern OLED screens, reducing heat and battery consumption when being used for extended periods of time. This may be considered "modernization", I guess?

**Comment by sanctimoniousape** [comment]
Ugh.  Does it work?  That's what matters.  I can't believe how many people complain about how something "looks" nowadays. The "Form Over Function" crowd has really taken over in recent years.

The only comment in this thread I agree with is the one mentioning OLED pure black usage, and that's more out of an ecological concern for conserving power & extending battery longevity wherever possible.  Otherwise, if you really give that much of a damn about "prettiness," then fork it and do it yourself. The dev (rightly, IMHO) doesn't feel the need to waste time on Google's regularly shifting guidelines on "design," and I can't blame them one bit.

^(Go ahead, bring on the hate.)

**Comment by VasilisKos** [comment]
> Ugh. Does it work? That's what matters. I can't believe how many people complain about how something "looks" nowadays. The "Form Over Function" crowd has really taken over in recent years. It's a **_tool,_** not a showpiece.
> 
> The only comment in this thread I agree with is the one mentioning OLED pure black usage, and that's more out of an ecological concern for conserving power & extending battery longevity wherever possible. Otherwise, if you really give that much of a damn about "prettiness," then fork it and do it yourself. The dev (rightly, IMHO) doesn't feel the need to waste time on Google's regularly shifting guidelines on "design," and I can't blame them one bit.
> 
> _(Go ahead, bring on the hate.)_

Who talked about "form over function"? How about "firm AND function" like hundreds of apps do? Isn't that an option for you?

**Comment by ThatOneCalculator** [comment]
Bringing it back to my original comment: 

> would help it feel cohesive with the rest of a modern Android phone.

I think there's a merit to trying to make sure users have a somewhat consistent visual and interactive experience across the applications on their device. Having 80-90% look like they're modern and the rest be somewhere in-between 2000s and 2010s design just makes things feel cheap and unpolished.

**Comment by woheller69** [comment]
Just do it. If it looks nice, I might accept a PR :-)

**Comment by ThatOneCalculator** [comment]
I would, but alas I'm a web developer.

**Comment by sanctimoniousape** [comment]
> > Ugh. Does it work? That's what matters. I can't believe how many people complain about how something "looks" nowadays. The "Form Over Function" crowd has really taken over in recent years. It's a **_tool,_** not a showpiece.
> > The only comment in this thread I agree with is the one mentioning OLED pure black usage, and that's more out of an ecological concern for conserving power & extending battery longevity wherever possible. Otherwise, if you really give that much of a damn about "prettiness," then fork it and do it yourself. The dev (rightly, IMHO) doesn't feel the need to waste time on Google's regularly shifting guidelines on "design," and I can't blame them one bit.
> > _(Go ahead, bring on the hate.)_
> 
> Who talked about "form over function"? How about "firm AND function" like hundreds of apps do? Isn't that an option for you?

If the developer so wishes to spend the time, that's obviously an option. However the app shows what it is intended to do, and the developer obviously feels they have more important things to do than to "make it pretty."  If I were in their shoes, I'd feel the same - especially since Google keeps moving the goalposts on design. 

Again, it's a freaking ***tool,*** not an art exhibit.  Pretty is nice, but hardly a priority over or even equal in importance proper function.  It's open source, so you have the option to contribute patches to make it pretty, or fork it if you prefer.  Don't badger the developer who has plainly said it's not a priority for them.

**Comment by OmlineEditor** [comment]
an example of how it can be done more modernly

![Image](https://github.com/user-attachments/assets/34ddf3be-b424-4065-b99c-ab04ae8affd2)

**Comment by ThatOneCalculator** [comment]
That's possibly the least modern design I've seen in a while.
